FogCreek/Glitch-Community
# Snow screwdriver: Show a meaningful error message when users try to use google translate

## Links
* https://snow-screwdriver.glitch.me/
* https://glitch.manuscript.com/f/cases/3328299/

## GIF/Screenshots:
🎥 http://recordit.co/zoKpU0nHiZ 🎥 

## Changes:
* detects when a user is attempting to use google translate to read our page and tries to show a meaningful error message. TIL that google translate has a nifty feature where it'll run your code through an iframe and try to translate all the text there. Unfortunately it seems as though this sort of behavior is not currently set up to deal with single page apps [like react apps](https://github.com/ReactTraining/history/issues/198), and to get this to work we'd have to do server side rendering (or just handle translation ourselves). Because it runs the code through an iframe, a lot of our relative links in react are broken and they throw cross origin errors like these: https://sentry.io/organizations/fog-creek-software/issues/885677384/?project=1246508&query=is%3Aunresolved making the website not translated and broken/unusable . Fortunately if we redirect the user in the iframe in this situation to glitch.com rather than to translate.google.com/abunchofstuff/glitch.com stuff goes back to working order. Also if we put a fallback message in our index.ejs file it gets translated nicely for us. 

## How To Test:
* go to https://translate.google.com/translate?sl=en&tl=es&u=snow-screwdriver.glitch.me see the translated message, see yourself get redirected to a working glitch.com
* to recreate the original errors: https://translate.google.com/translate?hl=&sl=en&tl=es&u=glitch.com&sandbox=1 (open the console and click on the fish on the upper left) notice how nothing happens.

## Feedback I'm looking for:
- anything I'm not thinking about? As an example iframes I know have various security issues/implications, is there anything not so great about this implementation I could have missed? Perhaps for example it's super bad to run community in an iframe altogether and we should detect this and throw an error message being like "don't run me an in an iframe silly"

## Things left to do before deploying:
- [x] want to double check this looks ok in a few other browsers,
- [x] also would like to check if there are other things like google translate that do the same thing
